src/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs

```diff
@@ -79,16 +89,47 @@ public bool CreateNoWindow
             set { _createNoWindow = value; }
         }
 
+        public StringDictionary EnvironmentVariables 
+        {
+            get 
+            {
+                                
+                if (_environmentVariables == null)
+                {
+#if PLATFORM_UNIX
+                    _environmentVariables = new StringDictionaryWithComparer(StringComparer.Ordinal);
+#else
+                    _environmentVariables = new StringDictionaryWithComparer(StringComparer.OrdinalIgnoreCase);
+#endif // PLATFORM_UNIX
+
+                    IEnumerator<KeyValuePair<string,string>> e = Environment.GetEnumerator();
```

Comment:
By specifying this as `IEnumerator<T>`, it's going to cause a box allocation of the dictionary's struct enumerator. It would be simpler to use `foreach` here.

```c#
foreach (KeyValuePair<string, string> pair in Environment)
{
    _environmentVariables.Add(pair.Key, pair.Value);
}
```